#!/usr/bin/env python
PACKAGE = "hough_circle"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_low", int_t, 0, "Hue low threshold", 115,0,255)
gen.add("h_high", int_t, 0, "Hue high threshold",125,0,255)
gen.add("s_low", int_t, 0, "Saturation low threshold", 180,0,255)
gen.add("s_high", int_t, 0, "Saturation high threshold",255,0,255)
gen.add("v_low", int_t, 0, "Value low threshold", 30,0,255)
gen.add("v_high", int_t, 0, "Value high threshold",255,0,255)

gen.add("max_radius",int_t,0,"Maximum radius of circle detected",150,0,150)
gen.add("min_radius",int_t,0,"Minimum radius of circle deteccted",20,0,40)
gen.add("binary_thres",int_t,0,"Threshold of the binary image",50,0,250)
gen.add("image_stream", bool_t, 0, "Enable/disable streaming image", True)
gen.add("debugging", bool_t, 0, "Enable/disable debugging mode", False)


exit(gen.generate(PACKAGE, "hough_circle", "Threshold"))
